dist/steps/run_tests.sh

```diff
@@ -7,6 +7,65 @@
 UNITY_PROJECT_PATH="$GITHUB_WORKSPACE/$PROJECT_PATH"
 echo "Using project path \"$UNITY_PROJECT_PATH\"."
 
+#
+# Create an empty project for testing if in package mode
+#
+
+if [ "$PACKAGE_MODE" = "true" ]; then
+  echo "Running tests on a Unity package rather than a Unity project."
+
+  echo ""
+  echo "###########################"
+  echo "#    Package Folder       #"
+  echo "###########################"
+  echo ""
+
+  ls "$UNITY_PROJECT_PATH"
+  echo ""
+
+  PACKAGE_JSON_PATH="$UNITY_PROJECT_PATH/package.json"
+  if [ ! -f "$PACKAGE_JSON_PATH" ]; then
+      echo "Unable to locate package.json from the given package path. Aborting..."
+      exit 1
+  fi
+
+  PACKAGE_NAME=$(cat "$PACKAGE_JSON_PATH" | jq -r ".name")
+
+  if [ -z $PACKAGE_NAME ]; then 
+    echo "Unable to parse package name from package.json. Aborting..."
+    exit 1
+  fi
+
+  echo "Package name found: ${PACKAGE_NAME}"
+
+  echo "Creating an empty Unity project to add the package to."
+
+  unity-editor \
+    -batchmode \
+    -createProject "./TempProject" \
```

Comment:
Since `TempProject` is used in multiple places, it might make sense to add it to a descriptive env variable (like `EMPTY_PROJECT` or `PACKAGE_TESTING_PROJECT`).

Which would also make this line self documenting, where:
```bash
  UNITY_PROJECT_PATH="./TempProject"

  # becomes

  UNITY_PROJECT_PATH="$PACKAGE_TESTING_PROJECT"
```